src/Raven.Server/Commercial/SetupInfo.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Raven.Client.Documents.Operations;
+using Raven.Client.Util;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Commercial
+{
+    public class SecuredSetupInfo
+    {
+        public License License { get; set; }
+        public string Email { get; set; }
+        public string Domain { get; set; }
+        public string Challenge { get; set; }
+        public List<NodeInfo> NodeSetupInfos { get; set; }
+        
+        public DynamicJsonValue ToJson()
+        {
+            return new DynamicJsonValue
+            {
+                [nameof(License)] = License.ToJson(),
+                [nameof(Email)] = Email,
+                [nameof(Domain)] = Domain,
+                [nameof(Challenge)] = Challenge,
+                [nameof(NodeSetupInfos)] = NodeSetupInfos.Select(node => node.ToJson()).ToArray()
+            };
+        }
+
+        public class NodeInfo
+        {
+            public string NodeTag { get; set; }
+            public string ServerUrl { get; set; }
+            public string PublicServerUrl { get; set; }
+            public int Port { get; set; }
+            public string Hostname { get; set; }
+            public string Certificate { get; set; }
+            public List<string> Ips { get; set; }
+
+
+            public DynamicJsonValue ToJson()
+            {
+                return new DynamicJsonValue
+                {
+                    [nameof(NodeTag)] = NodeTag,
+                    [nameof(ServerUrl)] = ServerUrl,
+                    [nameof(PublicServerUrl)] = PublicServerUrl,
+                    [nameof(Port)] = Port,
+                    [nameof(Hostname)] = Hostname,
+                    [nameof(Certificate)] = Certificate,
+                    [nameof(Ips)] = Ips.ToArray(),
+                };
+            }
+        }
+    }
+
+    public class UnsecuredSetupInfo
+    {
+        public string ServerUrl { get; set; }
+        public string PublicServerUrl { get; set; }
+
+        public DynamicJsonValue ToJson()
+        {
+            return new DynamicJsonValue
+            {
+                [nameof(ServerUrl)] = ServerUrl,
+                [nameof(PublicServerUrl)] = PublicServerUrl
+            };
+        }
+    }
+    
+    public class ClaimDomainInfo
+    {
+        public License License { get; set; }
+        public string Domain { get; set; }
+
+        public DynamicJsonValue ToJson()
+        {
+            return new DynamicJsonValue
+            {
+                [nameof(License)] = License.ToJson(),
+                [nameof(Domain)] = Domain
+            };
+        }
+    }
+
+    public class RegistrationResult
+    {
+        public RegistrationStatus Status { get; set; }
+
```

Comment:
Need to add additional message property here if so we can show an error to the user